src/trio/testing/_raises_group.py

```diff
@@ -307,28 +325,60 @@ def __init__(
         self,
         exception: type[E] | Matcher[E] | E,
         *other_exceptions: type[E] | Matcher[E] | E,
-        strict: bool = True,
+        allow_unwrapped: bool = False,
+        flatten_subgroups: bool = False,
         match: str | Pattern[str] | None = None,
         check: Callable[[BaseExceptionGroup[E]], bool] | None = None,
+        strict: None = None,
     ):
         self.expected_exceptions: tuple[type[E] | Matcher[E] | E, ...] = (
             exception,
             *other_exceptions,
         )
-        self.strict = strict
+        self.flatten_subgroups: bool = flatten_subgroups
+        self.allow_unwrapped = allow_unwrapped
         self.match_expr = match
         self.check = check
         self.is_baseexceptiongroup = False
 
+        if strict is not None:
+            warnings.warn(
+                DeprecationWarning(
+                    "`strict=False` has been replaced with `flatten_subgroups=True`"
+                    " with the introduction of `allow_unwrapped` as a parameter."
+                ),
+                stacklevel=2,
+            )
```

Comment:
mostly just me being confused about the existence of `TrioDeprecationWarning` (hence #2992). One weird part with `warn_deprecated` is you have to know what the next version bump will be, but I'm guessing I should mark it as `0.25.1`.
But I also don't think this one is worthy of an official extended period of supporting it, it's a trivial change to use `flatten_subgroups=True` instead and this is explicitly a temporary helper until it's merged into pytest, so it's barely worth having a DeprecationWarning at all. It's not used at runtime either, so it can only break tests. But I guess `TrioDeprecationWarning` doesn't have an explicit promise of how long it will be kept deprecated anyhow